UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -1397,38 +1396,26 @@ public virtual UnityEngine.Texture GetTexture(int textureIndex)
                     }
                 }
 
-                if (source.filterMode == desiredFilterMode && source.wrapMode == desiredWrapMode)
-                {
-                    _assetCache.TextureCache[textureIndex].Texture = source;
-                }
-                else
+                int sourceId = GetTextureSourceId(texture);
+                GLTF.Schema.Image image = _gltfRoot.Images[sourceId];
+                if(_assetCache.ImageCache[sourceId] != null)
                 {
-                    var unityTexture = UnityEngine.Object.Instantiate(source);
-                    unityTexture.filterMode = desiredFilterMode;
-                    unityTexture.wrapMode = desiredWrapMode;
-
-                    _assetCache.TextureCache[textureIndex].Texture = unityTexture;
+                    if(_assetCache.ImageCache[sourceId].filterMode != desiredFilterMode ||
+                       _assetCache.ImageCache[sourceId].wrapMode != desiredWrapMode)
+                    {
+                        // Note it is possible for a gltf file to have multiple textures with different sampling modes referencing the same image
+                        // Consider modifing ConstructImage to return a texture and either reusing the current texture from the TextureCache or 
+                        // creating a new one based on if the sampling mode matches
+                        Debug.LogWarning("Multiple textures with same image but different samplers, last loaded will win");

```

Comment:
The previous code had the functionality to return textures with differing samplers